### PR TITLE
Helm

### PR DESCRIPTION
**Pull request by ArshamTeymouri**
https://github.com/paritytech/devops/issues/1436

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
+
+.publish-refs:                     &publish-refs
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "web"
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+    - if: $CI_COMMIT_REF_NAME == "master"
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/
```
[the remark this answers is not in the document]
Comment (ArshamTeymouri) [reply .gitlab-ci.yml:52]: this is only for test purposes, I will remove this line when mirroring is configured by CI team, just want to make sure the pipeline works fine before merging.
all so all other job templates that are commented with `#fix` will be turned actual jobs by removing the `.` from the beginning of their names.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/ 
+
+# Run deploy to production only on tag
+.production-refs:                  &production-refs
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "web"
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/
```
[the remark this answers is not in the document]
Comment (ArshamTeymouri) [reply .gitlab-ci.yml:60]: same as above.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  script:
+    - yarn
+    - yarn lint
+
+test:
+  stage:                           test
+  <<:                              *kubernetes-env
+  <<:                              *common-refs
+  script:
+    - yarn test
```
Comment (joao-paulo-parity) [inline .gitlab-ci.yml:95]: does not exist for this project

File: CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#   which can be everywhere.
+# - Multiple owners are supported.
+# - Either handle (e.g, @github_user or @github_org/team) or email can be used. Keep in mind,
+#   that handles might work better because they are more recognizable on GitHub,
+#   you can use them for mentioning unlike an email.
+# - The latest matching rule, if multiple, takes precedence.
+
+# CI
+/.gitlab-ci.yml @paritytech/ci @joao-paulo-parity
+/helm/ @ArshamTeymouri @joao-paulo-parity
```
Comment (joao-paulo-parity) [inline CODEOWNERS:21]: I've added `@paritytech/devops` in the repository settings, feel free to assign the `/helm` folder to them

**Comment by joao-paulo-parity** [comment]
Since @alvicsam is out of office, and there are a few minor comments left, and I want to start trying the deployment this week, I'll address the remaining comments and get this merged soon

File: helm/values.yaml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    #   - ALL
+    # readOnlyRootFilesystem: true
+    # runAsNonRoot: true
+    # runAsUser: 1000
+    # runAsGroup: 1000
+  
+  service:
+    type: ClusterIP
+    port: 80
+    containerPort: 3000
```
[the remark this answers is not in the document]
Comment (joao-paulo-parity) [reply helm/values.yaml:65]: can't be done according to @ArshamTeymouri on DMs

> for every deployment, the containerPort should be specified explicitly.
> There's no way to reuse a key in values.yaml file.

File: .gitlab-ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Quintinlouw36) [reply .gitlab-ci.yml:60]: > same as above.

I totally agree on that...

**Pull request merged**